fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos').then(response => response.json()).then((todos) => {
+      this.setState({ todos });
+    })
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo),
+    })
+      .then(response => response.json())
+      .then(todos => {
+        let newList = this.state.todos;
```

Comment:
While this will work, one thing you do not want to do is make another reference to the original array, as that will end up mutating state. Instead you want to make a new copy and all the todo to that. Here is a useful article on the best ways to do that.  https://www.samanthaming.com/tidbits/35-es6-way-to-clone-an-array/